lua/entities/gmod_wire_expression2/core/custom/prop.lua

```diff
@@ -772,10 +772,39 @@ e2function void propDeleteAll()
 	self.data.spawnedProps = {}
 end
 
+--------------------------------------------------------------------------------
 
-__e2setcost(10)
+__e2setcost(1)
+e2function number entity:propIsDupeable()
+	return (this.DoNotDuplicate==nil and not duplicator.IsAllowed(this:GetClass()) or this.DoNotDuplicate) and 0 or 1
+end
 
---------------------------------------------------------------------------------
+local entMarkedNoDupeByPlayer = {}
```

Comment:
Avoid using a table because it will leak memory. Maybe just add a E2_MarkedNoDupe var onto the entity itself.